tesseract-core/src/lib.rs

```diff
@@ -196,6 +196,72 @@ impl Schema {
         Ok((sql, header))
     }
 
+    /// Convert user parameters into required default member cuts based on cube defintion.
+    ///
+    /// Given a cube and user supplied Query parameters and a boolean for negate mode, this function will:
+    ///
+    /// 1. Use the get_dims_for_default_member to determine which dimensions are not involved in
+    /// a cut or drilldown for the query and are therefore candidates for default member filtering logic
+    /// 2. It will read out the default_member strings from relevant hierarchies and parse them into cuts
+    /// 3. Returns a list of the cuts that should be added to the query based on the default members
+    ///
+    /// If the function runs in negate mode, it will add or remove a ~ from the front of the default member
+    /// cut string to flip the logic in order to exclude the default member from being returned in queries.
+    fn build_default_member_cuts(&self, schema_cube: &Cube, query: &Query, negate: bool) -> Box<Vec<Cut>> {
+        let target_dims = self.get_dims_for_default_member(schema_cube, query, negate);
+        let result = target_dims.iter().map(|dim| {
+            let target_hierarchy_name = match &dim.default_hierarchy {
+                Some(hierarchy_name) => hierarchy_name,
+                None => &dim.hierarchies.get(0).unwrap().name
+            };
+            let hierarchy_obj = &dim.hierarchies.iter()
+                .find(|h| &h.name == target_hierarchy_name).expect("bad hierarchy unpacking");
+            let default_member = &hierarchy_obj.default_member;
+            match default_member {
+                Some(val) => {
+                    let mut new_cut_str: String = val.to_string();
+                    if negate {
+                        let first_ch = new_cut_str.chars().next().expect("Expected at least one character in default member");
+                        new_cut_str = match first_ch {
+                            '~' => new_cut_str[1..].to_string(),
+                            _ => format!("~{}", new_cut_str)
+                        }
+
+                    }
+                    Cut::from_str(&new_cut_str)
+                },
+                None => Err(format_err!("Bad default member"))
+            }
+        }).filter_map(Result::ok).collect();
```

Comment:
Oh, I realized finally... if you filter map here, then all of the errors will be silently filtered out, instead of having errors propagate upward and stopping the iterator. I'm thinking that you didn't want to do that, since you explicitly set a `format_err` above.

Reply:
Yup - going to change this a bit so that the method returns Result<Vec<Cut>, Error>